health/interface.go

```diff
@@ -0,0 +1,13 @@
+package health
+
+// diagnosable is an internal interface, which external components can implement in order to be "diagnosed" by the health service
+type diagnosable interface {
+	Diagnose(deep bool)
```

Comment:
check if nil?

Reply:
Golang strikers again, so yes, I added the check :)

```
package main

import (
	"fmt"
)

type fooInterface interface {
	Foo()
}

type C struct {
}

func (c *C) Foo() {
}

func main() {
	c := (*C)(nil)
	do(c)
}

func do(o interface{}) {
	casted, ok := o.(fooInterface)
	fmt.Println(casted, ok) // <nil> true
}
```
:exploding_head:

Reply:
:)))